azext_iot/iothub/providers/helpers/edge_device_config.py

```diff
@@ -0,0 +1,577 @@
+
+EDGE_CONFIG_SCRIPT_HEADERS = """
+# This script will attempt to configure a pre-installed iotedge as a nested node.
+# It must be run as sudo, and will modify the ca
+
+device_id="{}"
+cp config.toml /etc/aziot/config.toml
+"""
+EDGE_CONFIG_SCRIPT_HOSTNAME = """
+# ======================= Set Hostname =======================================
+
+read -p "Enter the hostname to use: " hostname
+if [ -z "$hostname" ]
+then
+    echo "Invalid hostname $hostname"
+    exit 1
+fi
+
+sed -i "s/{{HOSTNAME}}/$hostname/" /etc/aziot/config.toml
+"""
+EDGE_CONFIG_SCRIPT_PARENT_HOSTNAME = """
+# ======================= Set Parent Hostname =======================================
+
+read -p "Enter the parent hostname to use: " parent_hostname
+if [ -z "$parent_hostname" ]
+then
+    echo "Invalid parent hostname $parent_hostname"
+    exit 1
+fi
+
+sed -i "s/{{PARENT_HOSTNAME}}/$parent_hostname/" /etc/aziot/config.toml
+"""
+EDGE_CONFIG_SCRIPT_CA_CERTS = f"""
+# ======================= Install nested root CA =======================================
+if [ -f /etc/os-release ]
+then
+        . /etc/os-release
+        if [[ "$NAME" == "Common Base Linux Mariner"* ]];
+        then
+                cp {EDGE_ROOT_CERTIFICATE_FILENAME} /etc/pki/ca-trust/source/anchors/{EDGE_ROOT_CERTIFICATE_FILENAME}.crt
+                update-ca-trust
+        else
+                cp {EDGE_ROOT_CERTIFICATE_FILENAME} /usr/local/share/ca-certificates/{EDGE_ROOT_CERTIFICATE_FILENAME}.crt
+                update-ca-certificates
+        fi
+else
+        cp {EDGE_ROOT_CERTIFICATE_FILENAME} /usr/local/share/ca-certificates/{EDGE_ROOT_CERTIFICATE_FILENAME}.crt
+        update-ca-certificates
+fi
+
+systemctl restart docker
+
+# ======================= Copy device certs  =======================================
+cert_dir="/etc/aziot/certificates"
+mkdir -p $cert_dir
+cp "{EDGE_ROOT_CERTIFICATE_FILENAME}" "$cert_dir/{EDGE_ROOT_CERTIFICATE_FILENAME}"
+cp "$device_id.full-chain.cert.pem" "$cert_dir/$device_id.full-chain.cert.pem"
+cp "$device_id.key.pem" "$cert_dir/$device_id.key.pem"
+"""
+EDGE_CONFIG_SCRIPT_HUB_AUTH_CERTS = """
+# ======================= Copy hub auth certs  =======================================
+cert_dir="/etc/aziot/certificates"
+mkdir -p $cert_dir
+cp "$device_id.hub-auth-cert.pem" "$cert_dir/$device_id.hub-auth-cert.pem"
+cp "$device_id.hub-auth-key.pem" "$cert_dir/$device_id.hub-auth-key.pem"
+"""
+EDGE_CONFIG_SCRIPT_APPLY = """
+# ======================= Read User Input =======================================
+iotedge config apply -c /etc/aziot/config.toml
+
+echo "To check the edge runtime status, run 'iotedge system status'. To validate the configuration, run 'sudo iotedge check'"
+"""
+
+EDGE_SUPPORTED_OS_LINK = "https://aka.ms/iotedge-supported-systems"
+EDGE_LINUX_TUTORIAL_LINK = "https://aka.ms/iotedge-provision-linux-device"
+EDGE_WINDOWS_TUTORIAL_LINK = "https://aka.ms/iotedge-provision-windows"
+
+DEVICE_README = f"""
```

Comment:
should this be a separate file rather than a constant?

Reply:
lots of these constants are scripts which make me question if they should be files instead

Reply:
Had a brief conversation about this before you left re: storing files in the extension, and then reading / writing them to device configs, or storing them as a string in code so we can just write to new files without reading the existing ones.

@digimaun curious if you have any thoughts here? I'm unsure of implications of including additional source files for these scripts. It would certainly make them easier to edit and format (and would probably save the linter some struggles) but is that worth the additional read steps every time we need to write these to a device bundle?